libs/deps/commerce.d.ts

```diff
@@ -0,0 +1,316 @@
+    }) => Promise<string | void>;
+
+    type providePriceOptions = (
+      element: HTMLInlinePriceElement,
+      options: Record<string, any>
+    ) => void;
+
+    type reset = () => void;
+
+    interface Instance {
+      readonly checkout: Checkout.Client;
+      readonly defaults: Defaults;
+      readonly ims: Ims.Client;
+      readonly providers: {
+        /**
+         * Registers price options provider - a function accepting
+         * price options object and modifying it as needed in specific context.
+         */
+        price(provider: providePriceOptions): () => void;
+      };
+      readonly literals: Literals;
+      readonly settings: Checkout.Settings & Wcs.Settings;
+      readonly wcs: Wcs.Client;
+    }
+
+    interface Literals {
+      price: Record<string, string>;
+    }
+
+    interface MiloConfig {
+      commerce?: Partial<
+        Record<keyof Checkout.Settings | keyof Wcs.Settings, any>
+      >;
+      env?: {
+        name: string;
+      };
+      locale?: {
+        ietf?: string;
+        prefix?: string;
+      };
+    }
+
+    interface Settings {
+      country: string;
+      env: Env;
+      language: string;
+      locale: string;
+    }
+
+    interface HTMLPlaceholderMixin {
+      init(): void;
+      onceSettled(): Promise<HTMLPlaceholderMixin>;
+      render(overrides?: Record<string, any>): void;
+      toggleFailed(version: number, error?: Error): boolean;
+      togglePending(): number;
+      toggleResolved(version: number): boolean;
+    }
+
+    interface HTMLCheckoutLinkElement extends HTMLPlaceholderMixin, HTMLAnchorElement {
+      renderOffers(offers: Wcs.Offer[], overrides?: Record<string, any>): void;
+    }
+
+    interface HTMLInlinePriceElement extends HTMLPlaceholderMixin, HTMLSpanElement {
+      new(): HTMLInlinePriceElement;
+      renderOffer(offer: Wcs.Offer, overrides?: Record<string, any>): void;
+    }
+
+    module Checkout {
+      type buildUrl = (options: Options) => string;
+      type Workflow = CheckoutType;
+
+      interface Client {
+        /**
+         * Returns checkout url string for given product offers and checkout options.
+         */
+        buildUrl: buildUrl;
+      }
+
+      interface Options extends CheckoutData {
+        checkoutPromoCode: string;
+        workflow: Workflow;
+      }
+
+      interface Settings extends Commerce.Settings {
+        checkoutClientId: string;
+        checkoutWorkflow: Workflow;
+        checkoutWorkflowStep: WorkflowStep;
+      }
+    }
+
+    module Ims {
+      interface Client {
+        /**
+         * A promise resolving with country code configured for current user in IMS.
+         */
+        get country(): Promise<string | void>;
+      }
+    }
+
+    module Internal {
+      interface Instance extends Omit<Commerce.Instance, 'providers'> {
+        providers: Providers;
+      }
+
+      interface Providers {
+        price: Set<Commerce.providePriceOptions>
+      }
+    }
+
+    module Log {
```

Comment:
as discussed offline, looks like this module is not commerce specific, as well as `debug` query parameter / variable. I would love not to have a commerce specific logging system, and we should probably discuss how best to reuse existing or integrate that if we need more than existing. 
cc @chrischrischris @auniverseaway